# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/angle/checkout/src/common/PackedEnums.cpp
# Commit: 13282d7a47a5
# Full Hash: 13282d7a47a57a73be8d060efd9645b4d642f03e
# Author: Miko Mynttinen <mikokm@gmail.com>
# Date: 2019-09-08 09:46:52
# Regressor Bug: 1578576
# File Overlap Count: 4
# Description:
#   Bug 1578576 - Part 2: Update to ANGLE 3865 r=jgilbert
#   
#   Depends on D44561
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D44579
# ==============================================================================

diff -r 3064469c073d -r 13282d7a47a5 gfx/angle/checkout/src/common/PackedEnums.cpp
--- a/gfx/angle/checkout/src/common/PackedEnums.cpp	Sat Sep 07 13:36:46 2019 +0000
+++ b/gfx/angle/checkout/src/common/PackedEnums.cpp	Sat Sep 07 13:38:36 2019 +0000
@@ -38,6 +38,8 @@
             return TextureType::_2DMultisampleArray;
         case TextureTarget::_3D:
             return TextureType::_3D;
+        case TextureTarget::InvalidEnum:
+            return TextureType::InvalidEnum;
         default:
             UNREACHABLE();
             return TextureType::InvalidEnum;
@@ -168,6 +170,114 @@
     }
 }
 
+std::ostream &operator<<(std::ostream &os, PrimitiveMode value)
+{
+    switch (value)
+    {
+        case PrimitiveMode::LineLoop:
+            os << "GL_LINE_LOOP";
+            break;
+        case PrimitiveMode::Lines:
+            os << "GL_LINES";
+            break;
+        case PrimitiveMode::LinesAdjacency:
+            os << "GL_LINES_ADJACENCY";
+            break;
+        case PrimitiveMode::LineStrip:
+            os << "GL_LINE_STRIP";
+            break;
+        case PrimitiveMode::LineStripAdjacency:
+            os << "GL_LINE_STRIP_ADJANCENCY";
+            break;
+        case PrimitiveMode::Points:
+            os << "GL_POINTS";
+            break;
+        case PrimitiveMode::TriangleFan:
+            os << "GL_TRIANGLE_FAN";
+            break;
+        case PrimitiveMode::Triangles:
+            os << "GL_TRIANGLES";
+            break;
+        case PrimitiveMode::TrianglesAdjacency:
+            os << "GL_TRIANGLES_ADJANCENCY";
+            break;
+        case PrimitiveMode::TriangleStrip:
+            os << "GL_TRIANGLE_STRIP";
+            break;
+        case PrimitiveMode::TriangleStripAdjacency:
+            os << "GL_TRIANGLE_STRIP_ADJACENCY";
+            break;
+        default:
+            os << "GL_INVALID_ENUM";
+            break;
+    }
+    return os;
+}
+
+std::ostream &operator<<(std::ostream &os, DrawElementsType value)
+{
+    switch (value)
+    {
+        case DrawElementsType::UnsignedByte:
+            os << "GL_UNSIGNED_BYTE";
+            break;
+        case DrawElementsType::UnsignedShort:
+            os << "GL_UNSIGNED_SHORT";
+            break;
+        case DrawElementsType::UnsignedInt:
+            os << "GL_UNSIGNED_INT";
+            break;
+        default:
+            os << "GL_INVALID_ENUM";
+            break;
+    }
+
+    return os;
+}
+
+std::ostream &operator<<(std::ostream &os, VertexAttribType value)
+{
+    switch (value)
+    {
+        case VertexAttribType::Byte:
+            os << "GL_UNSIGNED_BYTE";
+            break;
+        case VertexAttribType::Fixed:
+            os << "GL_FIXED";
+            break;
+        case VertexAttribType::Float:
+            os << "GL_FLOAT";
+            break;
+        case VertexAttribType::HalfFloat:
+            os << "GL_HALF_FLOAT";
+            break;
+        case VertexAttribType::Int:
+            os << "GL_INT";
+            break;
+        case VertexAttribType::Int2101010:
+            os << "GL_INT_10_10_10_2";
+            break;
+        case VertexAttribType::Short:
+            os << "GL_SHORT";
+            break;
+        case VertexAttribType::UnsignedByte:
+            os << "GL_UNSIGNED_BYTE";
+            break;
+        case VertexAttribType::UnsignedInt:
+            os << "GL_UNSIGNED_INT";
+            break;
+        case VertexAttribType::UnsignedInt2101010:
+            os << "GL_UNSIGNED_INT_10_10_10_2";
+            break;
+        case VertexAttribType::UnsignedShort:
+            os << "GL_UNSIGNED_SHORT";
+            break;
+        default:
+            os << "GL_INVALID_ENUM";
+            break;
+    }
+    return os;
+}
 }  // namespace gl
 
 namespace egl